### DISCUSSION TITLE
Source code inclusions require highlightjs?

### DISCUSSION THREAD

**Discussion by pitosalas** [hakimel/reveal.js#3351] [Plugins]
First of all, please tell me if there's a different place for me to post a question like this. Is there a forum or a discord channel?

I have lots of source code in my slides and I have been using highlightjs. Highlightjs does not have a feature for line numbers. And yet I have seen examples of revealjs pages with source code highlighted.

How is that done? A different code highlight package? Is there a sample or some doc on how to do that? My ideal code block in a presentation would have line numbers and smart copy/paste (where the line numbers are not copied.) 

Thank you!

**Comment by Martinomagnifico** [comment]
Hi Pito, you are correct that this is not the right place. “Discussions” would be more fitting for questions like this. As for a short answer (and not hard to find): [https://revealjs.com/code/](https://revealjs.com/code/).

**Reply by pitosalas** [reply]
Yes, I did find that and see that. The confusion is that highlight.js says that it doesn't do line numbers: https://highlightjs.readthedocs.io/en/latest/line-numbers.html . So when I saw line numbers on one of the pages I thought maybe you were using a different/seperate package for that :)

**Reply by hakimel** [reply]
Hi @pitosalas — we're using this highlight.js plugin to add line numbers: https://github.com/wcoder/highlightjs-line-numbers.js

**Comment by Martinomagnifico** [comment]
I’ll move this to discussions.

**Comment by Martinomagnifico** [comment]
And for a copy functionality, you could try a plugin I made: [Copycode](https://github.com/Martinomagnifico/reveal.js-copycode).

**Reply by pitosalas** [reply]
Yes I saw that too, thanks!
